src/project/qml/MuseScore/Project/AudioGenerationSettings.qml

```diff
@@ -113,12 +121,25 @@ RadioButtonGroup {
     Component {
         id: numberOfSavesComp
 
-        Row {
+        RowLayout {
             id: numberOfSavesItem
 
-            spacing: 0
+            width: parent.width
+            spacing: 12
+
+            //: `<span>%n</span>` will be replaced with a number input field.
+            //: Text before it will appear before that number field, text after will appear after the field.
+            readonly property string text: qsTrc("project/save", "Every: <span>%n</span> saves", "", settingsModel.numberOfSaves)
```

Comment:
Very confusing piece of code. Why do we need it?

Reply:
It's maybe indeed a bit complicated, I'll explain.

For the translators, it is very important that the text "Every: %n saves" is one big string, instead of split over multiple strings. This has to do with different word order in different languages, of course.

Thus, what we do, is making it one string, and replacing the "%n" with a number input field. Concretely, this means we need to split the string and put the part before %n before the spinbox, and the part after "%n" after the spinbox.

Recall that a translatable string containing "%n" allows translators to give multiple translations for singular / plural / whatever forms their language has. 
For example:
1 Save
2 Saves
(English is a bad example in this case because it has only two forms and in this particular case the singular form will never occur, but you get the idea. There are languages with way more complex rules, so we really need to support that.)

However, this comes with a small implication: to make it work correctly, we _must_ specify the value of "n" as the fourth argument to `qsTrc`. This is necessary to:
- make sure that `lupdate` correctly finds the string and identifies it as a "%n" string
- make sure that we retrieve the correct form of the translation (singular/plural/...)

But it implies that the "n" is already filled in in the string, so we get "Every: 2 saves" instead of "Every: %n saves". Normally that's great, but in this case it's a tiny bit inconvenient, since it makes it harder to determine where to split the string.

So we do a trick, and surround the number with something we can recognize but won't harm either, namely `<span>` and `</span>`.

Now, we can easily use a regular expression to split the string.

Reply:
(I added some documentation in the code too)

Reply:
No. 
This is a difficult task and it is better not to waste your time on it.
Not only your time, but also the time of translators, the time of developers who will read this code and a lot of documentation for it. The code should be simple.

It is better to let the user see 1 saves, I think he will not be offended much.

Reply:
I honestly don't see what's so complicated about it. It took me literally a few seconds to come up with this idea. And correct grammar _is_ important. In my opinion, you can put as many effort in an app as you want, but if it contains language mistakes, it just looks unprofessional. 

Anyway, opinions aside - if we choose a less perfect solution, with %1 instead of %n, it won't be much less complicated. The only simplification is that we can replace the regular expression with `.split("%1")`. Is that worthwhile?

Reply:
I'm also not totally convinced the plural is worth worrying about in this case, but if we do want to do this then I think the following would be sufficient to explain what is going on to developers and translators:

```QML
//: `<span>%n</span>` will be replaced with a number input field for the user to pick a number.
//: Text before `<span>%n</span>` will appear before the number field, text after will appear after the field.
readonly property string text: qsTrc("project/save", "Every: <span>%n</span> save(s)", "", settingsModel.numberOfSaves)
// We need `<span>` tags to be able to split the string after it has been translated and `%n` replaced with a number.
```

Notice I replaced "saves" with "save(s)" to give a hint about the plural.

Presumably this should also be done with "every %1 measures" above.

Reply:
>I honestly don't see what's so complicated about it. It took me literally a few seconds to come up with this idea
 
 This doesn't mean that your solution is easy to understand for other developers. It looks incredibly overcomplicated
 
 I like Peter's suggestion. Just change the unit text to "save(s)". But tbh, I doubt very much that we need to do anything with this text at all

Reply:
Okay, I resorted to using simply %1, to avoid the scary regular expression. Let's hope nobody will notice the imperfection :)

But FWIW, we _must_ make it one string and splitting it. I'm not doing this for academical reasons, but because translators already reported that they can't work with these separate strings. 

Also, if the two words are in two different strings, translators have no clue that they form one phrase together, or what they actually mean (unless we start adding enormous comments and hints etc.). If you see the whole phrase, it's totally clear and unambiguously translatable. If I were a translator and saw just the word "Saves" for example, I could come up with many possible Dutch translations with totally different meanings and I would have no idea which one is correct.

Reply:
I agree it should be one string. The `<span>` idea was actually quite an elegant solution. I think it was just the large comment above made it look more complicated than it really was.